Assets/_Game/Scripts/Weapons/Weapon.cs

```diff
@@ -1,31 +1,39 @@
-using UnityEngine;
+﻿using UnityEngine;
+using com.ootii.Messages;
 
 public class Weapon : MonoBehaviour {
-  public GameObject bulletPrefab;
-  public float fireRate = 0.5f;
-  public float shootingRange = 10f;
+  private WeaponDataLoader weaponDataLoader {
+    get => WeaponDataLoader.Instance;
+  }
+  public Transform attackPoint;
   private bool isFacingRight = true;
   private float fireTimer;
 
   void Update() {
     fireTimer += Time.deltaTime;
 
-    if (fireTimer >= fireRate) {
+    if (fireTimer >= weaponDataLoader.firerate) {
       fireTimer = 0f;
 
       FindAndFireAtTarget();
     }
   }
 
+  void Start() {
+    MessageDispatcher.AddListener("Right" , Flip);
+    MessageDispatcher.AddListener("Left" , Flip);
+  }
+
+
   void FindAndFireAtTarget() {
     GameObject[] enemies = GameObject.FindGameObjectsWithTag(Constants.Tag_Enemy);
```

Comment:
Remove GameObject.FindGameObjectsWithTag(Constants.Tag_Enemy)
Add spawned enemies to a list then access them at here